templates/search/search_clients_last_seen_v1/query.sql

```diff
@@ -0,0 +1,164 @@
+WITH
+  _derived_search_cols AS (
+    SELECT
+      CASE
+          WHEN engine IS NULL THEN 'none'
+          WHEN STARTS_WITH(engine, 'google') THEN 'google'
+          WHEN STARTS_WITH(engine, 'ddg') OR STARTS_WITH(engine, 'duckduckgo') THEN 'ddg'
+          WHEN STARTS_WITH(engine, 'bing') THEN 'bing'
+          WHEN STARTS_WITH(engine, 'yandex') THEN 'yandex'
+          ELSE 'other'
+      END AS short_engine,
+      COALESCE(organic, 0) + COALESCE(sap, 0) + COALESCE(unknown, 0) + COALESCE(tagged_sap, 0) + COALESCE(tagged_follow_on, 0) AS total_searches,
+      COALESCE(tagged_sap, 0) + COALESCE(tagged_follow_on, 0) AS tagged_searches,
+      COALESCE(ad_click, 0) AS ad_click,
+      COALESCE(search_with_ads, 0) AS search_with_ads,
+      * EXCEPT (ad_click, search_with_ads)
+    FROM
+      search_clients_daily_v7
+    WHERE
+      submission_date_s3 = @submission_date
+      and sample_id = "84"
+  ),
+
+  _derived_engine_searches AS (
+    SELECT
+      STRUCT(
+        short_engine AS key,
+        STRUCT(
+          total_searches,
+          tagged_searches,
+          ad_click,
+          search_with_ads
+        ) AS value
+      ) AS engine_searches,
+      * EXCEPT (short_engine)
+    FROM
+      _derived_search_cols
+  ),
+
+  _grouped AS (
+    -- Here we get a single row per-client, containing
+    -- info from each engine, as well as overall info
+    SELECT
+      -- Grouping columns
+      client_id,
+      CAST(sample_id AS INT64) AS sample_id,
+
+      -- Dimensional data
+      udf_mode_last(ARRAY_AGG(country)) AS country,
+      udf_mode_last(ARRAY_AGG(app_version)) AS app_version,
+      udf_mode_last(ARRAY_AGG(distribution_id)) AS distribution_id,
+      udf_mode_last(ARRAY_AGG(locale)) AS locale,
+      udf_mode_last(ARRAY_AGG(search_cohort)) AS search_cohort,
+      udf_mode_last(ARRAY_AGG(addon_version)) AS addon_version,
+      udf_mode_last(ARRAY_AGG(os)) AS os,
+      udf_mode_last(ARRAY_AGG(channel)) AS channel,
+      udf_mode_last(ARRAY_AGG(profile_creation_date)) AS profile_creation_date,
+      udf_mode_last(ARRAY_AGG(default_search_engine)) AS default_search_engine,
+      udf_mode_last(ARRAY_AGG(default_search_engine_data_load_path)) AS default_search_engine_data_load_path,
+      udf_mode_last(ARRAY_AGG(default_search_engine_data_submission_url)) AS default_search_engine_data_submission_url,
+      udf_mode_last(ARRAY_AGG(profile_age_in_days)) AS profile_age_in_days,
+      udf_mode_last(ARRAY_AGG(active_addons_count_mean)) AS active_addons_count_mean,
+      udf_mode_last(ARRAY_AGG(user_pref_browser_search_region)) AS user_pref_browser_search_region,
+      udf_mode_last(ARRAY_AGG(os_version)) AS os_version,
+
+      -- User activity data
```

Comment:
Ben, are these usage aggregate appropriate for `search_clients_daily`? The `SUM` fields are especially concerning because we don't want to double count anything. What this assumes is that now rows in `search_clients_daily` share any common e.g. `tab_open_event_count_sum` values, which may not be true if a single session used multiple engines.

Reply:
hm, these fields have particularly weird behavior, as I was going to suggest using `FIRST()` here instead of `SUM()`, however in `search_clients_daily`, for a client-day, these values are not always the same (for the same engine):

https://console.cloud.google.com/bigquery?pli=1&project=moz-fx-data-derived-datasets&folder&organizationId=442341870013&j=bq:US:bquxjob_6da07000_16e04a3130a&page=queryresults

Reply:
Looking over the code for `search_clients_daily` I don't think there's anyway we can get guaranteed correct values for these per-client. The problem is that some of them may have occurred in the same session, but some may not have, and we don't know which are which.

Reply:
AFAIK, this only affects these non search-related fields:

```sql
SUM(tab_open_event_count_sum) AS tab_open_event_count_sum,
SUM(active_hours_sum) AS active_hours_sum,
SUM(subsession_hours_sum) AS subsession_hours_sum,
SUM(sessions_started_on_this_day) AS sessions_started_on_this_day
```

Couldn't we treat these aggregations separately, which would require a join on `client_id`? This is something we'd want to do for `search_clients_daily` as well I'm thinking. It would duplicate the fields for clients with multiple `engine` or `source` fields (i.e. a client with 4 active hours will report 4 active hours for every row in scd), and we'd have to aggregate using `FIRST()` for downstream analysis.  Its not pretty, but I'm not sure if the current implementation has a useful interpretation.

Reply:
Right, we're on the same page here. Honestly my thought was to simply elide them, since by themselves they aren't useful, and aggregating them together isn't viable.